dod-advana/gamechanger-web
# Task/updated re

## Description
This updates the existing responsibility explorer to have a new default document view as well as an admin panel for verifying any updates made by users. Also include a SQL file to be run for deployment: v104.0_update_responsibilities

## Tickets
* UOT-120470
* UOT-120473
* UOT-120478
* UOT-121898
* UOT-124006

## Testing Instructions
* The PDF viewer does not work locally. You will either need to test it from a dev site it is deployed on or comment out the iframe in GCResponsibilityDocumentExplorer.js to test with out the PDF viewer functionality.
* Navigate to the responsibility explorer to land on new document view
* Test filtering capabilities
* Test pagination
* Expand Documents and organizations and click on responsibilities. Verify meta data is loaded in right table. 
* Navigate back and forth between chart view and document view with the toggle in the top right. Have filters applied as you toggle. 
* In the document view, edit and reject responsibilities and their organizations using the buttons in the meta table on the right. Highlight the new text on the pdf highlighter now in the middle panel and save it
* Navigate to the admin panel and to the Responsibility Updates page
* Navigate through different updates. Update meta data for each responsibility should appear on the right. Clicking on an updates' text should jump to and highlight that update in the PDF.
* Edit an update with the Edit button in the meta table. 
* Reject an update
* Accept updates and ensure the updated text appears in the RE results
* The global Enter key listener for the search bar was also removed. Verify the search bar still works properly as far as key listeners go.
